src/main/java/com/pinterest/secor/common/aws/AssumeRoleAndDefaultCredentialsProviderChain.java

```diff
@@ -0,0 +1,72 @@
+package com.pinterest.secor.common.aws;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.amazonaws.auth.*;
+import com.amazonaws.auth.profile.ProfileCredentialsProvider;
+import com.amazonaws.auth.profile.internal.*;
+import com.amazonaws.auth.profile.internal.securitytoken.STSProfileCredentialsServiceLoader;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import static com.amazonaws.profile.path.AwsProfileFileLocationProvider.DEFAULT_CONFIG_LOCATION_PROVIDER;
+import static com.amazonaws.profile.path.AwsProfileFileLocationProvider.DEFAULT_CREDENTIALS_LOCATION_PROVIDER;
+
+public class AssumeRoleAndDefaultCredentialsProviderChain extends AWSCredentialsProviderChain {
+
+  private static final Log log = LogFactory.getLog(AssumeRoleAndDefaultCredentialsProviderChain.class);
+
+  private static final AssumeRoleAndDefaultCredentialsProviderChain INSTANCE
+    = new AssumeRoleAndDefaultCredentialsProviderChain();
+
+  public AssumeRoleAndDefaultCredentialsProviderChain() {
+    super(getDefaultCredentials(),
+      new SystemPropertiesCredentialsProvider(),
+      new ProfileCredentialsProvider(),
+      new EC2ContainerCredentialsProviderWrapper(),
+      WebIdentityTokenCredentialsProvider.builder().build());
+  }
+
+  public static AssumeRoleAndDefaultCredentialsProviderChain getInstance() {
+    return INSTANCE;
+  }
+
+  private static AWSCredentialsProvider getDefaultCredentials() {
+    log.info("Using AssumeRoleAndDefaultCredentialsProviderChain");
+    final String profileName = "profile " + AwsProfileNameLoader.INSTANCE.loadProfileName();
```

Comment:
when deployed as service this assumes we have awscli setup, I think we can treat it as one of those secret in config?